config/routes.rb

```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get 'users/:user_id/posts', to: 'posts#index', :as => 'user_posts'
+  get 'users/:user_id/posts/:id', to: 'posts#show', :as => 'user_post'
+  get '/users',   to: 'users#index'
+  get "/users/:id", to: "users#show", :as => 'user'
```

Comment:
- [ ] [Optional] It would be better professionally if you add `users#index` as a root route and list your routes like that:

# config/routes.rb
```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end

  root "users#index"
```

Reply:
Amazing, will do that
thanks for the suggestion.